src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types/Todo';
+import React, { ChangeEvent, useState } from 'react';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const getPreparedTodos = () => {
+  return todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId) || null,
+  }));
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(getPreparedTodos());
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [titleInputError, setTitleInputError] = useState(false);
+  const [userSelectError, setUserSelectError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setSelectedUserId(0);
+    setTitleInputError(false);
+    setUserSelectError(false);
+  };
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    setTitleInputError(!title);
+    setUserSelectError(!selectedUserId);
+
+    if (!title && !selectedUserId) {
+      return;
+    }
```

Comment:
The condition `if (!title && !selectedUserId)` checks both `title` and `selectedUserId` together. This means that if either `title` or `selectedUserId` is missing, the function will not return early, potentially leading to incorrect error handling. Consider using separate conditions for each field to ensure proper validation.